test_kubeyaml.py

```diff
@@ -41,31 +52,20 @@ def strip(s):
                    strats.just('__') | \
                    strats.integers(min_value=1, max_value=5).map(lambda n: '-' * n)
 
-image_tags = strats.from_regex(r"^[a-z][\w.-]{0,127}$").map(strip)
-
-@composite
-def image_components(draw):
-    bits = draw(strats.lists(
-        elements=strats.text(alphabet=alphanumerics, min_size=1),
-        min_size=1, max_size=16))
-    s = bits[0]
-    for c in bits[1:]:
-        sep = draw(image_separators)
-        s = s + sep + c
-    return s
-
 host_segments = strats.just([]) | hostnames.map(lambda x: [x])
-image_names = strats.builds(lambda host, cs: '/'.join(host + cs),
-                            host_segments,
-                            strats.lists(elements=image_components(),
-                                         min_size=1, max_size=6))
 
+# This results in realistic image refs, we use a min_size of two for
+# test cases that make use of a hostname as the Docker image ref spec
+# has a limitation on images with a hostname that requires it to have
+# at least two elements if the hostname is not localhost.
+exact_image_names = strats.builds('/'.join, strats.lists(elements=image_components(), min_size=2, max_size=6))
```

Comment:
See the comment below on the use of exact_names

Reply:
It's a minor thing, but I believe this can go back to being `min_size=1`, since it represents _only_ the image repository part. (i.e., if exact image refs were used below, they would have to include the possibility of hostnames as well.)

Reply:
I do not think that would work, as this would result in

```
<hostname> / <image component> : <tag>
```

which _may_ generate an image ref like `0/a:a` which would bring us back to https://github.com/squaremo/kubeyaml/pull/12#discussion_r295277284

Reply:
My point earlier was that this is a problem with the hostname production, and not with the image components. `a:a` is a valid image name.

Reply:
To work around the quirk in the spec, `hostnames` does now contain `localhost` as single element hostname plus hostnames with 2-6 components.